lib/MySQL_Session.cpp

```diff
@@ -244,10 +341,53 @@ int MySQL_Session::handler() {
 				break;
 
 			case WAITING_CLIENT_DATA:
+				if (pkt.size==(0xFFFFFF+sizeof(mysql_hdr))) {
+					// we are handling a multi-packet
+					switch (client_myds->DSS) {
+						case STATE_SLEEP:
+							client_myds->DSS=STATE_SLEEP_MULTI_PACKET;
+							break;
+						case STATE_SLEEP_MULTI_PACKET:
+							break;
+						default:
+							assert(0);
+							break;
+					}
+				}
 				switch (client_myds->DSS) {
+					case STATE_SLEEP_MULTI_PACKET:
+						if (client_myds->multi_pkt.ptr==NULL) {
+							// not initialized yet
+							client_myds->multi_pkt.ptr=pkt.ptr;
+							client_myds->multi_pkt.size=pkt.size;
+						} else {
+							PtrSize_t tmp_pkt;
```

Comment:
Very elegant handling of this multi-packet usecase! Much more elegant than my previous implementation